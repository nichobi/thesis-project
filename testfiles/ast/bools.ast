Expression ::= Term;

abstract Term;
True  : Term;
False : Term;
Or    : Term ::= Left:Term Right:Term;
Zero  : Term;

abstract Type;
Bool : Type;
Nat : Type;
