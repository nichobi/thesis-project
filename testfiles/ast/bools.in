[T-True]
-----------
True : Bool

[T-False]
------------
False : Bool

[T-Or]
left : Bool,
right : Bool
---------------
Or(left, right) : Bool

[T-Zero]
--------
Zero : Int

