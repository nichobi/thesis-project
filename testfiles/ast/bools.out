RuleSet
  List
    Rule Name="T-True"
      HasType
        Identifier ID="true"
        Identifier ID="Bool"
      List
    Rule Name="T-False"
      HasType
        Identifier ID="false"
        Identifier ID="Bool"
      List
    Rule Name="T-Or"
      HasType
        RuleTerm RuleName="or"
          List
            Identifier ID="a"
            Identifier ID="b"
        Identifier ID="Bool"
      List
        HasType
          Identifier ID="a"
          Identifier ID="Bool"
        HasType
          Identifier ID="b"
          Identifier ID="Bool"
    Rule Name="T-Zero"
      HasType
        Identifier ID="0"
        Identifier ID="Int"
      List
