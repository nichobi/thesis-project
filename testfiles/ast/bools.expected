RuleSet
  List
    Rule Name="T-True"
      HasType
        Value ID="True"
        Value ID="Bool"
      List
    Rule Name="T-False"
      HasType
        Value ID="False"
        Value ID="Bool"
      List
    Rule Name="T-Or"
      HasType
        Function ID="Or"
          List
            Value ID="left"
            Value ID="right"
        Value ID="Bool"
      List
        HasType
          Value ID="left"
          Value ID="Bool"
        HasType
          Value ID="right"
          Value ID="Bool"
    Rule Name="T-Zero"
      HasType
        Value ID="Zero"
        Value ID="Int"
      List
